pkg/api/api.go

```diff
@@ -445,3 +445,12 @@ const (
 	// UserCancel user cancelled compose up, we are stopping containers
 	UserCancel
 )
+
+// GetDefaultImageName computes the default image name for a service, used to tag built images
+func GetDefaultImageName(service types.ServiceConfig, projectName string) string {
```

Comment:
I'd propose renaming this to `GetImageNameOrDefault` since it reads `service.Image` if populated and then falls back to producing one. Particularly now that it's exported 😉